src/lib/print.c

```diff
@@ -495,33 +495,26 @@ char *vp_aprints_type(TALLOC_CTX *ctx, PW_TYPE type)
  * @param out Where to write the string.
  * @param outlen Length of output buffer.
  * @param vp to print.
+ * @param raw_value if true, the raw value is printed and not the enumerated attribute value
  * @return the length of data written to out, or a value >= outlen on truncation.
  */
-size_t vp_prints_value_json(char *out, size_t outlen, VALUE_PAIR const *vp)
+size_t vp_prints_value_json(char *out, size_t outlen, VALUE_PAIR const *vp, bool raw_value)
 {
 	char const	*q;
 	size_t		len, freespace = outlen;
+	bool		raw = raw_value || (!vp->da->flags.has_tag && !vp->da->flags.has_value);
```

Comment:
I don't entirely understand this line - why are we checking for has_value here? We previously were checking for has_tag only.

I think this is causing my environment to always have `raw` as true.

I assumed FreeRADIUS only ever sent the raw value until coming across this PR in my travels on the chunked stuff - so am trying to figure out why I don't ever see enumerated values.

Reply:
So that the integer value is printed where the attribute doesn't have a set of enumeration values I guess...

Reply:
Yeah - that makes sense. I'm confused because in my environment I don't see enumeration values ever being printed, only the integer. I.e. `Acct-Status-Type=Start` going in comes out as `{"Acct-Status-Type":{"type":"integer","value":[1]}}`

I've tested on 3.0.19 through 3.0.23 with very basic config and see the same behaviour though, so, I guess it's not related to this change, and I have some more digging to do.

Thanks!